libvlcpp: init 0.1.0-unstable-2024-02-04

fliiiix opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] ~~Tested basic functionality of all binary files (usually in `./result/bin/`)~~ header only
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fliiiix committed:
libvlcpp: Init 0.1.0-unstable-2024-02-04

FliegendeWurst reviewed (commented):
Output looks reasonable
```
/nix/store/7zp08vxz61xd1nrw6pnhm71q2n7kzncz-libvlcpp-0.1.0-unstable-2024-02-04
├── include
│   └── vlcpp
│       ├── common.hpp
│       ├── Dialog.hpp
│       ├── Equalizer.hpp
│       ├── EventManager.hpp
│       ├── Instance.hpp
│       ├── Internal.hpp
│       ├── MediaDiscoverer.hpp
│       ├── Media.hpp
│       ├── MediaLibrary.hpp
│       ├── MediaList.hpp
│       ├── MediaListPlayer.hpp
│       ├── MediaPlayer.hpp
│       ├── Picture.hpp
│       ├── RendererDiscoverer.hpp
│       ├── structures.hpp
│       └── vlc.hpp
├── lib
│   └── pkgconfig
│       └── libvlcpp.pc
└── nix-support
    └── propagated-build-inputs
```
Does the package really require `vlc`? The pkg-config file only mentions libvlc.

fliiiix commented:
@FliegendeWurst no idea what made me think it needed vlc i rebased to master and removed the vlc dependency and retested it again and it works without

SuperSandro2000 commented on pkgs/by-name/li/libvlcpp/package.nix:
```diff
@@ -0,0 +1,39 @@
+  version = "0.1.0-unstable-2024-02-04";
+
+  src = fetchFromGitLab {
+    owner = "videolan";
+    repo = "libvlcpp";
+    domain = "code.videolan.org";
```
```suggestion
    domain = "code.videolan.org";
    owner = "videolan";
    repo = "libvlcpp";
```

SuperSandro2000 commented on pkgs/by-name/li/libvlcpp/package.nix:
```diff
@@ -0,0 +1,39 @@
+    repo = "libvlcpp";
+    domain = "code.videolan.org";
+    rev = "44c1f48e56a66c3f418175af1e1ef3fd1ab1b118";
+    sha256 = "sha256-nnS4DMz/2VciCrhOBGRb1+kDbxj+ZOnEtQmzs/TJ870=";
```
```suggestion
    hash = "sha256-nnS4DMz/2VciCrhOBGRb1+kDbxj+ZOnEtQmzs/TJ870=";
```

SuperSandro2000 commented on pkgs/by-name/li/libvlcpp/package.nix:
```diff
@@ -0,0 +1,39 @@
+  doCheck = true;
+
+  meta = {
+    description = "A header-only, C++ bindings for the libvlc crossplatform multimedia API";
```
```suggestion
    description = "Header-only C++ bindings for the libvlc crossplatform multimedia API";
```

SuperSandro2000 commented on pkgs/by-name/li/libvlcpp/package.nix:
```diff
@@ -0,0 +1,39 @@
+    sha256 = "sha256-nnS4DMz/2VciCrhOBGRb1+kDbxj+ZOnEtQmzs/TJ870=";
+  };
+
+  propagatedBuildInputs = [
```
Do we really need to propagate libvlc?

fliiiix replied:
i would say yes without it it's relatively useless

SuperSandro2000 commented on pkgs/by-name/li/libvlcpp/package.nix:
```diff
@@ -0,0 +1,39 @@
+    sha256 = "sha256-nnS4DMz/2VciCrhOBGRb1+kDbxj+ZOnEtQmzs/TJ870=";
+  };
+
+  propagatedBuildInputs = [
+    libvlc
+  ];
+  nativeBuildInputs = [
+    pkg-config
+    autoreconfHook
+  ];
```
```suggestion
  nativeBuildInputs = [
    pkg-config
    autoreconfHook
  ];

  propagatedBuildInputs = [
    libvlc
  ];
```
